cocotb/_build_libs.py

```diff
@@ -0,0 +1,446 @@
+    )
+
+    #
+    #  libgpi
+    #
+    libgpi = Extension(
+        os.path.join("cocotb", "libs", sim_define.lower(), "libgpi"),
+        define_macros=[("LIB_EXT", _get_lib_ext_name()), ("SINGLETON_HANDLES", "")],
+        include_dirs=[include_dir],
+        libraries=["cocotbutils", "gpilog", "cocotb", "stdc++"],
+        sources=[
+            os.path.join(share_lib_dir, "gpi", "GpiCbHdl.cpp"),
+            os.path.join(share_lib_dir, "gpi", "GpiCommon.cpp"),
+        ],
+        extra_link_args=_extra_link_args("libgpi"),
+        extra_compile_args=_extra_cxx_compile_args,
+    )
+
+    #
+    #  simulator
+    #
+    libsim = Extension(
+        os.path.join("cocotb", "libs", sim_define.lower(), "simulator"),
+        include_dirs=[include_dir],
+        libraries=[_get_python_lib_link(), "cocotbutils", "gpilog", "gpi"],
+        library_dirs=python_lib_dirs,
+        sources=[os.path.join(share_lib_dir, "simulator", "simulatormodule.c")],
+        extra_compile_args=_extra_cc_compile_args,
+    )
+
+    return [libcocotbutils, libgpilog, libcocotb, libgpi, libsim]
+
+
+def _get_vpi_lib_ext(
+    include_dir, share_lib_dir, sim_define, extra_lib=[], extra_lib_dir=[]
+):
+    libvpi = Extension(
+        os.path.join("cocotb", "libs", sim_define.lower(), "libvpi"),
+        define_macros=[("VPI_CHECKING", "1")] + [(sim_define, "")],
+        include_dirs=[include_dir],
+        libraries=["gpi", "gpilog"] + extra_lib,
+        library_dirs=extra_lib_dir,
+        sources=[
+            os.path.join(share_lib_dir, "vpi", "VpiImpl.cpp"),
+            os.path.join(share_lib_dir, "vpi", "VpiCbHdl.cpp"),
+        ],
+        extra_link_args=["-Wl,-rpath,$ORIGIN"],
+        extra_compile_args=_extra_cxx_compile_args,
+    )
+
+    return libvpi
+
+
+def _get_vhpi_lib_ext(
+    include_dir, share_lib_dir, sim_define, extra_lib=[], extra_lib_dir=[]
+):
+    libcocotbvhpi = Extension(
+        os.path.join("cocotb", "libs", sim_define.lower(), "libcocotbvhpi"),
+        include_dirs=[include_dir],
+        define_macros=[("VHPI_CHECKING", 1)] + [(sim_define, "")],
+        libraries=["gpi", "gpilog", "stdc++"] + extra_lib,
+        library_dirs=extra_lib_dir,
+        sources=[
+            os.path.join(share_lib_dir, "vhpi", "VhpiImpl.cpp"),
+            os.path.join(share_lib_dir, "vhpi", "VhpiCbHdl.cpp"),
+        ],
+        extra_link_args=["-Wl,-rpath,$ORIGIN"],
+        extra_compile_args=_extra_cxx_compile_args,
+    )
+
+    return libcocotbvhpi
+
+
+def get_ext():
+
+    cfg_vars = distutils.sysconfig.get_config_vars()
+
+    if sys.platform == "darwin":
+        cfg_vars["LDSHARED"] = cfg_vars["LDSHARED"].replace("-bundle", "-dynamiclib")
+
+    share_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), "share"))
+    share_lib_dir = os.path.relpath(os.path.join(share_dir, "lib"))
+    share_def_dir = os.path.abspath(os.path.join(share_dir, "def"))
+    include_dir = os.path.relpath(os.path.join(share_dir, "include"))
```

Comment:
If you move these lines out of the function, then you'll be able to use `share_def_dir` in the `build_ext` subclass.